package.json

```diff
@@ -12,9 +14,24 @@
     "@testing-library/user-event": "^13.1.2",
     "@types/jest": "^26.0.21",
     "babel-jest": "^26.6.3",
+    "eslint": "^7.29.0",
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-plugin-import": "^2.23.4",
+    "eslint-plugin-jest": "^24.3.6",
+    "husky": "^4.0.0",
     "jest": "^26.6.3",
+    "lint-staged": "^11.0.0",
     "live-server": "^1.2.1",
     "msw": "^0.29.0",
     "prettier": "^2.3.0"
+  },
+  "lint-staged": {
+    "**/*.js": "yarn run lint"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged",
+      "pre-push": "yarn test"
```

Comment:
I might be in the minority here (and if so I'm happy to defer to majority opinion), but I find pre-commit/push hooks a bit annoying 😅. My editor lets me know when I am violating lint and/or style rules, and I run tests in watch mode while I develop. So in my case having extra checks run automatically is fairly redundant. That being said, if the total lint + test time took ~1s or less, it wouldn't really be a big deal, though that could change case as we add more tests.

Reply:
yeah i do the same. 

the hooks i am think will be the 'training' for other that don't (including one off folks who contribute). in essence it will save them a push (and to an extent commit) round trip. this is esp. considering we've never had it in the code base so far.

Reply:
@parthopdas sounds good to me, thanks for explaining!